### DISCUSSION TITLE
sub command to inspect quarto files or quarto projects?

### DISCUSSION THREAD

**Discussion by eitsupi** [quarto-dev/quarto-cli#856] [Q&A] [answered]
There are many yaml options that are automatically selected by quarto without explicitly stating them, or that are set by default (e.g. `engine`).
It might be useful to have a subcommand that displays the options selected as a result of quarto's scan of the document, without rendering it.
(I am imagining something like the `docker compose config` command.)

**Comment by mcanouil** [comment]
I was thinking of something similar, but for the website, i.e., in addition of having pieces of yaml a bit everywhere, it would be really useful to have some kind of reference card of all yaml options in one place on the website.

**Reply by cscheid** [reply]
We should improve the description of https://quarto.org/docs/reference/, but this page is a central point of access to our yaml documentation. It's partly generated by the YAML validation code, and so it should be reliable as a single source of truth for YAML options.

**Reply by mcanouil** [reply]
Thanks, I was thinking more of a table of contents / glossary to get quickly a glance at all the options, possibly with links to the appropriate sections in "reference".
In a way, the json code you provided in the main thread answers that.

Context: if you do not know what exists or not, it's hard to search for it. With the "reference" page, you have it all, but you have to go through it all and remember where the option you were interested was.

**Reply by cscheid** [reply]
Right. The thing is, the JSON in that answer is only a subset of the total options. We don't have a single page partly because, as of right now, we have yaml information on 437 options, and those are a subset of the options that control the behavior in Quarto :)

**Reply by mcanouil** [reply]
I suspected something like this.
Then maybe, not all, but only the one described in "reference".
It could be some kind of a new "overview" section in the reference page 🤔

**Comment by cscheid** [comment, marked as answer]
As a developer and user of quarto, I couldn't agree more! This is squarely on our radar. We don't currently advertise or document `quarto inspect` (see below for some reasons why), but it might help you here. It produces JSON that looks roughly like this:

```json
 {
  "quarto": {
    "version": "99.9.9"
  },
  "engines": [
    "markdown"
  ],
  "formats": {
    "html": {
      "execute": {
        "fig-width": 7,
        "fig-height": 5,
        "fig-format": "retina",
        "fig-dpi": 96,
        "error": false,
        "eval": true,
        "cache": null,
        "freeze": false,
        "echo": true,
        "output": true,
        "warning": true,
        "include": true,
        "keep-md": true,
        "keep-ipynb": false,
        "ipynb": null,
        "enabled": null,
        "daemon": null,
        "daemon-restart": false,
        "debug": false,
        "ipynb-filters": [],
        "engine": "markdown"
      },
      "render": {
        "keep-tex": false,
        "keep-yaml": false,
        "keep-source": false,
        "keep-hidden": false,
        "prefer-html": false,
        "output-divs": true,
        "output-ext": "html",
        "fig-align": "default",
        "fig-pos": null,
        "fig-env": null,
        "code-fold": "none",
        "code-overflow": "scroll",
        "code-link": false,
        "code-line-numbers": false,
        "code-tools": false,
        "tbl-colwidths": "auto",
        "merge-includes": true,
        "latex-auto-mk": true,
        "latex-auto-install": true,
        "latex-clean": true,
        "latex-max-runs": 10,
        "latex-makeindex": "makeindex",
        "latex-makeindex-opts": [],
        "latex-tlmgr-opts": [],
        "latex-input-paths": [],
        "latex-output-dir": null,
        "link-external-icon": false,
        "link-external-newwindow": false
      },
      "pandoc": {
        "standalone": true,
        "wrap": "none",
        "default-image-extension": "png",
        "to": "html",
        "output-file": "test-mermaid.html"
      },
      "language": {},
      "metadata": {
        "lang": "en",
        "fig-responsive": true,
        "quarto-version": "99.9.9"
      },
      "extensions": {
        "book": {
          "multiFile": true
        }
      }
    }
  },
  "resources": []
}
```

One of the main complications before fully supporting something like this is that some of the default behavior depends on which downstream tooling we are targeting. In order for this to be comprehensive, we would also need to document default options for eg. `knitr`, `xelatex`, `rmarkdown`, `jupyter`, and `pandoc`.

**Reply by eitsupi** [reply]
Thank you for the explanation.
I had assumed that such subcommands were common and was relieved to see that they were already implemented.
This is definitely useful in certain use cases. Thank you for developing this!
